### PR TITLE
Added provider filtering

### PR DESCRIPTION
**Pull request by farhatahmad**
## Description

Added provider filtering on the 4 main admin pages

### CODE REVIEW

File: spec/models/recording_spec.rb [hunk 1]
```diff
@@ -15,6 +15,22 @@
...
+    context 'with_provider' do
+      it 'only includes users with the specified provider' do
+        room1 = create(:room, user: create(:user, provider: 'greenlight'))
+        room2 = create(:room, user: create(:user, provider: 'test'))
+
+        create_list(:recording, 5, room: room1)
+        create_list(:recording, 5, room: room2)
+
+        recs = described_class.includes(:user).with_provider('greenlight')
+        expect(recs.count).to eq(5)
```
Comment (KH-Amir-TN) [inline spec/models/recording_spec.rb:28]: I think these can be merged into one if we do something like `expect(recs.pluck(:id)).to match_array(greenlight_recs.pluck(:id))`

File: spec/models/role_spec.rb [hunk 2]
```diff
@@ -12,6 +12,19 @@
...
 
+  describe 'scopes' do
+    context 'with_provider' do
+      it 'only includes users with the specified provider' do
+        create_list(:role, 5, provider: 'greenlight')
+        create_list(:role, 5, provider: 'test')
+
+        roles = described_class.with_provider('greenlight')
+        expect(roles.count).to eq(5)
+        expect(roles.pluck(:provider).uniq).to eq(['greenlight'])
```
Comment (KH-Amir-TN) [inline spec/models/role_spec.rb:23]: Same here,

File: spec/models/room_spec.rb [hunk 3]
```diff
@@ -3,7 +3,32 @@
...
+      it 'only includes users with the specified provider' do
+        user1 = create(:user, provider: 'greenlight')
+        user2 = create(:user, provider: 'test')
+
+        create_list(:room, 5, user: user1)
+        create_list(:room, 5, user: user2)
+
+        rooms = described_class.includes(:user).with_provider('greenlight')
+        expect(rooms.count).to eq(5)
+        expect(rooms.pluck(:provider).uniq).to eq(['greenlight'])
```
Comment (KH-Amir-TN) [inline spec/models/room_spec.rb:28]: Same as above,

File: spec/models/room_spec.rb [hunk 4]
```diff
@@ -3,7 +3,32 @@
...
 
 RSpec.describe Room, type: :model do
-  let!(:room) { create(:room) }
+  let(:room) { create(:room) }
+
+  describe 'validations' do
+    subject { create(:room) }
+
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to have_many(:recordings).dependent(:destroy) }
```
Comment (KH-Amir-TN) [inline spec/models/room_spec.rb:12]: I know this is irrelevant to this PR but to not forget,
In the future we may want to tirgger a recording delete logic (maybe in a job) that will remove recordings both on GL and on BBB.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -30,21 +30,34 @@
...
+      it 'only includes users with the specified provider' do
+        create_list(:user, 5, provider: 'greenlight')
+        create_list(:user, 5, provider: 'test')
 
-      it 'returns all users if input is empty' do
-        create_list(:user, 10)
-        expect(described_class.all.search('').pluck(:id)).to match_array(described_class.all.pluck(:id))
+        users = described_class.with_provider('greenlight')
+        expect(users.count).to eq(5)
+        expect(users.pluck(:provider).uniq).to eq(['greenlight'])
```
Comment (KH-Amir-TN) [inline spec/models/user_spec.rb:43]: Same,

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -30,21 +30,34 @@
...
       end
     end
   end
 
+  describe '#search' do
+    it 'returns the searched users' do
+      searched_users = create_list(:user, 5, name: 'Jane Doe')
+      create_list(:user, 5)
+      expect(described_class.all.search('jane doe').pluck(:id)).to match_array(searched_users.pluck(:id))
```
Comment (KH-Amir-TN) [inline spec/models/user_spec.rb:52]: Is the `all` required?

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -30,21 +30,34 @@
...
+    end
+
+    it 'returns all users if input is empty' do
+      create_list(:user, 10)
+      expect(described_class.all.search('').pluck(:id)).to match_array(described_class.all.pluck(:id))
```
Comment (KH-Amir-TN) [inline spec/models/user_spec.rb:57]: Same here,

**Review by KH-Amir-TN — DISMISSED** [review]

File: spec/models/recording_spec.rb (hunk 1 above)
Comment (farhatahmad) [reply spec/models/recording_spec.rb:28]: I prefer doing an explicit check to confirm that all providers are what we expect

Comment (farhatahmad) [reply spec/models/recording_spec.rb:28]: (in case something happens in the factory for some reason)

File: spec/models/user_spec.rb (hunk 6 above)
Comment (farhatahmad) [reply spec/models/user_spec.rb:52]: nope

**Pull request merged**